src/video/wayland/SDL_waylandvideo.c

```diff
@@ -617,10 +620,15 @@ static void display_handle_done(void *data,
         if (video->viewporter != NULL) {
             SDL_AddDisplayMode(dpy, &native_mode);
         } else {
-            /* ...if not, expose the integer scale desktop modes down to 1.0. */
+            /* ...if not, expose some smaller, integer scaled resolutions. */
             int i;
-            for (i = (int)driverdata->scale_factor - 1; i > 0; --i) {
+            const int base_w = desktop_mode.pixel_w /= (int)desktop_mode.display_scale;
```

Comment:
Are you intending to divide the pixel width here?

Reply:
Yes, it's intentional.  This is calculating lower, integer scaled resolutions to be made available should the desktop not support the viewport protocol and we have to fall back to the integer scaling path, and to calculate these, the 1x base pixel value is used.

Reply:
Can you split it out so it doesn't look like a typo?

Reply:
Actually this doesn't need to be `/=`, because it's re-assigned below, right?

Reply:
Fixed.